pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -175,6 +186,68 @@ func (s *Service) getOrCreateListener(openStackCluster *infrav1.OpenStackCluster
 	return listener, nil
 }
 
+func (s *Service) getOrUpdateAllowedCIDRS(openStackCluster *infrav1.OpenStackCluster, listener *listeners.Listener) error {
+	allowedCIDRs := []string{}
+
+	if len(openStackCluster.Spec.APIServerLoadBalancer.AllowedCIDRs) > 0 {
+		allowedCIDRs = append(allowedCIDRs, openStackCluster.Spec.APIServerLoadBalancer.AllowedCIDRs...)
+
+		if openStackCluster.Spec.Bastion.Enabled {
+			allowedCIDRs = append(allowedCIDRs, openStackCluster.Status.Bastion.FloatingIP, openStackCluster.Status.Bastion.IP)
+		}
+
+		if openStackCluster.Status.Network.Subnet.CIDR != "" {
+			allowedCIDRs = append(allowedCIDRs, openStackCluster.Status.Network.Subnet.CIDR)
+		}
+
+		if len(openStackCluster.Status.Network.Router.IPs) > 0 {
+			allowedCIDRs = append(allowedCIDRs, openStackCluster.Status.Network.Router.IPs...)
+		}
+	}
+
+	// validate CIDRs and convert any given IP into a CIDR
+	allowedCIDRs = validateIPs(allowedCIDRs)
+
+	// sort and remove duplicates
```

Comment:
Is this actually sorting the slice?

```suggestion
	// remove duplicates
```